### PR TITLE
Fix 143

### LINKED ISSUE #143: Option to preserve blank lines
I'd like to have an option such that fantomas do not adjust blank lines. I have my own blank lining convention, and fantomas sometimes makes the code too compact by deleting blank lines. So it would be nice if fantomas has an option to preserve blank lines as it is

### PR DESCRIPTION
**Pull request by s-trooper**
This closes #143, this PR preserve original end of lines, disables auto insert/remove of EOLs by Formatter. 
As it give control to line breaking to developer, it implicit solves #43,  #130, #132, #133? but that of course depends of expectations of issue creators.

### CODE REVIEW

**Comment by s-trooper** [comment]
Hi @stroborobo, thank you for the report. I have fixed the problem.

**Comment by nojaf** [comment]
Hi @s-trooper , we looked at your PR. Looks good! Could you elaborate on the ignored unit tests?
@dungpa could you perhaps review this one please?

**Comment by s-trooper** [comment]
Hi @ nojaf, i tested my RP by format the fantomas source,  the ignored tests are some problems I had, that prevent the source code from being compiled after formatting. Since those issues are not caused by my RP (some stuff is related to F# Compiler Services), I used the ignored tests to document them for later.

Probably I had to put these tests in a separate branch, should I do that? Or just ignore them when merging the RP.

File: src/Fantomas.Tests/PreserveEOLTests.fs [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+let ``array values``() =
+    formatSourceString false """
+let x = [ 1
+          2]
+let y = [| 3;4
+           5;6|]
+let z = [7; 8]
+    """ { config with SemicolonAtEndOfLine = false }
+    |> should equal """
+let x = [ 1;
```
Comment (dsyme) [inline src/Fantomas.Tests/PreserveEOLTests.fs]: Just curious: why is this adding a `;` when `SemicolonAtEndOfLine = false` is set?  I guess it is unrelated to this PR

**Comment by nojaf** [comment]
Hi @s-trooper , regarding the ignored tests. I'd prefer if they are left out of this PR. We could create separate issues if necessary.

**Comment by s-trooper** [comment]
@nojaf, i've removed ignored tests

**Pull request merged**